### PR TITLE
Object-Cache Update

### PR DESCRIPTION
**Pull request by katelynnbarlowe**
I've updated the drop-in based on the [Redis Cache version](https://raw.githubusercontent.com/rhubarbgroup/redis-cache/develop/includes/object-cache.php). I swapped out our variable in `wp_cache_init()`. I tested on one of my sites and everything _seems_ to work as it did before. The only thing I'm not sure about for this PR is the versioning/tagging - what should I update that to? 

It looks like we automatically compare versions and update the dropin in the `show_object_cache_dropin_updated_notice()` in `AdminNotices.php`.

### CODE REVIEW

**Comment by A5hleyRich** [comment]
@katelynnbarlowe this could do with a refresh: 

https://github.com/rhubarbgroup/redis-cache/releases/tag/2.0.22

File: drop-ins/object-cache.php [hunk 1]
```diff
@@ -150,9 +152,28 @@ function wp_cache_incr( $key, $offset = 1, $group = '' ) {
 function wp_cache_init() {
     global $wp_object_cache;
 
+    if ( ! defined( 'WP_REDIS_PREFIX' ) && getenv( 'SPINUPWP_CACHE_KEY_SALT' ) ) {
+        define( 'WP_REDIS_PREFIX', getenv( 'SPINUPWP_CACHE_KEY_SALT' ) );
+    }
+
+    if ( ! defined( 'WP_REDIS_SELECTIVE_FLUSH' ) && getenv( 'WP_REDIS_SELECTIVE_FLUSH' ) ) {
+        define( 'WP_REDIS_SELECTIVE_FLUSH', (bool) getenv( 'WP_REDIS_SELECTIVE_FLUSH' ) );
+    }
```
Comment (A5hleyRich) [inline drop-ins/object-cache.php:161]: This should always be `true` in Spinup https://github.com/deliciousbrains/spinupwp-plugin/commit/7cc2b8c68cb3ff886431bdfe8511657f6503ac67#diff-e176a90a3c73dc81c0254b1712e3a7e65b13375641c60a4e8836af1bbe8958ad

File: drop-ins/object-cache.php [hunk 2]
```diff
@@ -150,9 +152,28 @@ function wp_cache_incr( $key, $offset = 1, $group = '' ) {
...
+
+    // Backwards compatibility: map `WP_CACHE_KEY_SALT` constant to `WP_REDIS_PREFIX`.
+    if ( defined( 'WP_CACHE_KEY_SALT' ) && ! defined( 'WP_REDIS_PREFIX' ) ) {
+        define( 'WP_REDIS_PREFIX', WP_CACHE_KEY_SALT );
+    }
+
+    // Set unique prefix for sites hosted on Cloudways
+    if ( ! defined( 'WP_REDIS_PREFIX' ) && isset( $_SERVER['cw_allowed_ip'] ) )  {
+        define( 'WP_REDIS_PREFIX', getenv( 'HTTP_X_APP_USER' ) );
+    }
```
Comment (A5hleyRich) [inline drop-ins/object-cache.php:171]: We don't need this.

File: drop-ins/object-cache.php [hunk 3]
```diff
@@ -150,9 +152,28 @@ function wp_cache_incr( $key, $offset = 1, $group = '' ) {
...
+
+    if ( ! defined( 'WP_REDIS_SELECTIVE_FLUSH' ) && getenv( 'WP_REDIS_SELECTIVE_FLUSH' ) ) {
+        define( 'WP_REDIS_SELECTIVE_FLUSH', (bool) getenv( 'WP_REDIS_SELECTIVE_FLUSH' ) );
+    }
+
+    // Backwards compatibility: map `WP_CACHE_KEY_SALT` constant to `WP_REDIS_PREFIX`.
+    if ( defined( 'WP_CACHE_KEY_SALT' ) && ! defined( 'WP_REDIS_PREFIX' ) ) {
+        define( 'WP_REDIS_PREFIX', WP_CACHE_KEY_SALT );
+    }
```
Comment (A5hleyRich) [inline drop-ins/object-cache.php:166]: `WP_REDIS_PREFIX` and `WP_CACHE_KEY_SALT` needs reworking to handle all of these possibilities:

https://github.com/deliciousbrains/spinupwp-plugin/blob/develop/drop-ins/object-cache.php#L2001

**Review by A5hleyRich — CHANGES_REQUESTED** [review]
@katelynnbarlowe a few comments.

**Comment by katelynnbarlowe** [comment]
@A5hleyRich I _think_ I've gotten it where it needs to be. I added back the function that was there before to account for all of the different cases.

**Comment by katelynnbarlowe** [comment]
@A5hleyRich Latest updates have been brought in! It wasn't too much.

**Comment by A5hleyRich** [comment]
💯

**Pull request merged**